ghost/members-api/lib/members-api.js

```diff
@@ -264,7 +264,7 @@ module.exports = function MembersAPI({
             }
         }
 
-        const newMember = await users.create({name, email, labels, newsletters, attribution, geolocation});
+        const newMember = await users.create({name, email, labels, newsletters, attribution, geolocation, type});
```

Comment:
Im not sure about passing `type` to the repository here - `type` seems to be an API / request specific thing. Why do we need to pass this? Wouldn't `newsletters` be enough?

Reply:
Hmm...I passed `type` because I have added [these checks](https://github.com/TryGhost/Ghost/blob/637f5c927c292f30852763914e22cfa2c715f28f/ghost/members-api/lib/repositories/MemberRepository.js#L286) and we don't want this validation for the `sign-up` flow through portal.

Reply:
Is there not another way that we can achieve this without explicitly passing `type` to the repository? Can we not just check for the presence of `memberData.newsletters` and then validate the contents?

Reply:
we have memberData.newsletters in both `signup` and `subscribe` type. Let me think on this one again.

Reply:
whats the problem with them being in both flows? Is it because you are only wanting to validate in one of the flows? If so, what is the harm in validating in both flows?